src/libYARP_OS/include/yarp/os/impl/StreamConnectionReader.h

```diff
@@ -54,7 +54,8 @@ class YARP_OS_impl_API yarp::os::impl::StreamConnectionReader : public Connectio
         ref(YARP_NULLPTR),
         convertedTextMode(false),
         pushedIntFlag(false),
-        parentConnectionReader(YARP_NULLPTR)
+        parentConnectionReader(YARP_NULLPTR),
+        pushedInt(-1)
```

Comment:
This is causing several -Wreorder warnings on linux.
Please fix this so that they are initialized in the same order as they are declared (just inverting `pushedInt` and `parentConnectionReader` should do it).